upgrade/steps.go

```diff
@@ -1121,12 +1121,19 @@ var fetchLatestJavaGen = stepv2.Func03("Fetching latest Java Gen", func(ctx cont
 	return currentJavaGen, latestJavaGen.String(), true
 })
 
-var parseUpstreamProviderOrg = stepv2.Func11("Get UpstreamOrg from module version", func(ctx context.Context, upstreamMod module.Version) string {
+var parseUpstreamProviderOrg = stepv2.Func11E("Get UpstreamOrg from module version", func(ctx context.Context, upstreamMod module.Version) (string, error) {
 	// We expect tokens to be of the form:
 	//
 	//	github.com/${org}/${repo}/${path}
 	//
 	// The second chunk is the org name.
+
+	// Verify that the token is of valid format
+	goModRegexp := regexp.MustCompile("github.com/[a-zA-Z0-9-]*/[a-zA-Z0-9-]*")
```

Comment:
Q: do we need to make this less restrictive to allow underscores and other characters?

Reply:
Great question! I'm not aware of any repos that have those at this time. But the point here is to check for a complete module path so I don't see any reason not to add the.

For reference:

[Underscores are not allowed in TF module names](https://developer.hashicorp.com/terraform/language/style#module-repository-names)

Underscores are not allowed in GH org names (I couldn't find docs, but trying to set up an organization proved this), but [GitLab seems to allow it](https://docs.gitlab.com/ee/user/reserved_names.html#limitations-on-usernames-project-and-group-names), and allows dots as well.
Dots seem to be allowed in GitHub also, although GitHub substitutes those for dashes.